docs/source/extending/high-level.rst

```diff
@@ -119,6 +119,52 @@ the function's name will be mangled. To find out the mangled name of
 your function you can check the extension module's ``__pyx_capi__``
 attribute.
 
+Importing F2PY Functions
+--------------------------
+
+The function ``get_f2py_function_address`` obtains the address of a
+C function in a F2PY extension module. The address can be used to
+access the C function via a :func:`ctypes.CFUNCTYPE` callback, thus
+allowing use of the C function inside a Numba jitted function. For
+example, call scipy.interpolate.dfitpack.splev::
+
+    import numba as nb
+    import numpy as np
+    import ctypes
+    from scipy import interpolate
+    from numba.core.extending import get_f2py_function_address
+
+    dble_p=ctypes.POINTER(ctypes.c_double)
+    int_p =ctypes.POINTER(ctypes.c_longlong)
+
+    #Calling siganture
+    functype = ctypes.CFUNCTYPE(ctypes.c_void_p,
+                                dble_p,int_p,dble_p,int_p,dble_p,dble_p,int_p,int_p,int_p)
+
+    #Get function pointer and create ctypes function
+    func_ptr=get_f2py_function_address("scipy.interpolate.dfitpack","splev")
+    SPLEV = functype(func_ptr)
+
+You can access ``SPLEV`` from Numba in the following way::
+
+    from numba.cpython.unsafe.stack_arr import val_to_ptr,ptr_to_val
+
+    #float64[::1] -> ensure contiguous array, call np.ascontiguousarray or 
+    #np.asfortranarray instead to ensure to pass a pointer to a contiguous array
+    @nb.njit('float64[::1](float64[::1],float64[::1],int64,float64[::1],int64)')
+    def splev_wrapped(t,c,k,x,e):
+        y=np.empty(x.shape[0],dtype=np.float64)
+
+        n_arr=val_to_ptr(nb.int64(t.shape[0]))
+        k_arr=val_to_ptr(nb.int64(k))
+        m_arr=val_to_ptr(nb.int64(x.shape[0]))
+        e_arr=val_to_ptr(nb.int64(e))
+        ier_arr=val_to_ptr(nb.int64(0))
+
+        SPLEV(t.ctypes,n_arr,c.ctypes,k_arr,x.ctypes,
+            y.ctypes,m_arr,e_arr,ier_arr)
+        return y
+
```

Comment:
This may need adjusting with respect to suggested changes in this PR. I think it also makes sense to add this as a unit test in `numba.tests.doc_examples` and then render that straight into the docs, this will ensure that the example use case of this public API is unlikely to be silently broken. Suggest following something like this for writing the test (the `magictoken` comments indicate included region boundaries for the docs):
https://github.com/numba/numba/blob/3cc5e5e97664c65d92172b8f9401e835ce0825ca/numba/tests/doc_examples/test_literal_container_usage.py#L58-L90
and including in the docs like: https://github.com/numba/numba/blob/3cc5e5e97664c65d92172b8f9401e835ce0825ca/docs/source/reference/pysupported.rst?plain=1#L838-L846